### ISSUE TITLE
No house number quests displayed

### ISSUE THREAD

**Issue by Aschroet** [westnordost/StreetComplete#1369]
What is the reason why no house number quests are shown e. g. in this area?

https://www.openstreetmap.org/?mlat=#map=19/50.82700/10.96400

For me it seems to be a bug. But maybe i also missed something that needs to be added to OSM first. But what?

Level and roof shapes were/are displayed properly.

**Versions affected**
Android 8.0.0 / StreetComplete 11.0

**Comment by westnordost** [comment]
The building type needs to be specified first.

Am 5. April 2019 07:52:16 MESZ schrieb Aschroet <notifications@github.com>:
>What is the reason why no house number quests are shown e. g. in this
>area?
>
>https://www.openstreetmap.org/?mlat=#map=19/50.82700/10.96400
>
>For me it seems to be a bug. But maybe i also missed something that
>needs to be added to OSM first. But what?
>
>Level and roof shapes were/are displayed properly.
>
>**Versions affected**
>Android 8.0.0 / StreetComplete 11.0

**Comment by matkoniecz** [comment]
> The building type needs to be specified first.

And there is quest for that! So sooner or later quests for house number should appear.

The requirement for house type is necessary as otherwise StreetComplete would be asking about house number of sheds, garages and other buildings that typically have no house numbers.

**Comment by Aschroet** [comment]
Understand but look at these two objects:
https://www.openstreetmap.org/way/327637842
https://www.openstreetmap.org/way/182736594

Both have the has "building=house" but only the first one is shown in StreetComplete. It is only an example of many such cases. Btw, for the second object i already solved the quests roof shape, levels etc. till no others appeared. But the house number quest never has been shown.

**Comment by matkoniecz** [comment]
Have you tried zooming in? Maybe its quest marker is blocked by other nearby one?

If on a maximum zoom it is not displayed you may try to change order of quests in settings and move housenumber to top and press "scan for more quests" in menu (I just did it and quest for https://www.openstreetmap.org/way/182736594 appeared).

Also, https://www.openstreetmap.org/way/182736594/history was recently edited. Maybe StreetComplete fetched housenumbers quests in your area more than 6 days ago and is still using this data? You may use "invalidate cache" in option to throw away cached data and force a new download that will include your recent changes.

----

In general it should sooner or later appear on its own but it is currently not displayed due to some technical/performance/UX limitations. Solving nearby quests and waiting a bit should make it appear though one may force it to appear earlier.

**Comment by Aschroet** [comment]
Of course i tried zooming in and out before to find the quests. But the hint with "invalidate cache" helped. Now i see the house number quests. However, for a stupid end user as me it was a little bit frustrating not seeing all those quests in a row when i was on the spot. Maybe, this can be somehow solved.

**Comment by westnordost** [comment]
Thank you, matkoniecz

Am 6. April 2019 11:01:07 MESZ schrieb Mateusz Konieczny <notifications@github.com>:
>Have you tried zooming in? Maybe its quest marker is blocked by other
>nearby one?
>
>If on a maximum zoom it is not displayed you may try to change order of
>quests in settings and move housenumber to top and press "scan for more
>quests" in menu (I just did it and quest for
>https://www.openstreetmap.org/way/182736594 appeared).
>
>Also, https://www.openstreetmap.org/way/182736594/history was recently
>edited. Maybe StreetComplete fetched housenumbers quests in your area
>more than 6 days ago and is still using this data? You may use
>"invalidate cache" in option to throw away cached data and force a new
>download that will include your recent changes.
>
>----
>
>In general it should sooner or later appear on its own but it is
>currently not displayed due to some technical/performance/UX
>limitations. Solving nearby quests and waiting a bit should make it
>appear though one may force it to appear earlier.

**Comment by matkoniecz** [comment]
Maybe it would be a good idea to invalidate just house number quest once `building=yes` quest added house type that likely has address?

BTW, it is duplicate of #1119

**Comment by Aschroet** [comment]
Actually, the dependencies of the quests are important. If one quest changes the value that another quest relies on, this quest should be invalidated.

**Comment by laurentS** [comment]
Hi!
I'm having the same issue in my neighbourhood. I've been doing a few updates almost everyday and some quests refuse to show up, particularly house type / house number (this is purely subjective!). Here is an example: https://www.openstreetmap.org/way/201851160#map=19/51.59335/-0.19466
This house (as most in the neighbourhood) is tagged `building=yes`. Somehow the building type shows up for some, but not all. Same issue with housenumbers.
I've been in this area dozens of times over the past few months, so I am puzzled.
I've tried the actions suggested above, but somehow it doesn't fix it 100%. Would there be a way to "force download" quests for a very small area?

**Comment by westnordost** [comment]
So if this is neither because you did not zoom in enough, this will be the same issue as discussed in #1357: There are so many building type quests in the area, that the app decides to not download them all but only the first few thousand (per downloaded area) or so for performance reasons. This will be mitigated in the next version because in the future, the app will download smaller areas.
You can "force download" a small(ish) area by zooming in to the max and then hitting the "search for quests here" button.

**Comment by laurentS** [comment]
Ah ok, the limit makes sense. Reading through that issue, it seems pretty much the same problem as mine (I'm in London, so there are definitely tons of quests around me). Thanks for the super fast reply, and keep it up, your app is really well made! (this is coming from a fellow mapping app developer :)

**Comment by westnordost** [comment]
Oh cool, what app are you developing?

**Comment by laurentS** [comment]
https://github.com/mapswipe/mapswipe/ There's a major upgrade coming up soon, essentially a full rewrite. The app is used to help locate populated areas in humanitarian context, so we can then focus the work of mappers on those areas. If you've heard of the use of OSM in the current Ebola response in DRC for instance, mapswipe helped with that.
